### PR TITLE
PLANNER-2142 Make OptaPlanner 8 a part of kogito release pipeline

### PR DESCRIPTION
**Pull request by rsynek**
https://issues.redhat.com/browse/PLANNER-2142

Related PRs:
https://github.com/kiegroup/optaplanner/pull/976
https://github.com/kiegroup/kogito-examples/pull/402

### CODE REVIEW

File: Jenkinsfile.release [hunk 1]
```diff
@@ -48,13 +53,19 @@ pipeline {
     }
 
     environment {
-        KOGITO_CI_EMAIL_TO = credentials('KOGITO_CI_EMAIL_TO')
+        //KOGITO_CI_EMAIL_TO = credentials('KOGITO_CI_EMAIL_TO')
+        KOGITO_CI_EMAIL_TO = credentials('RSYNEK_EMAIL')
```
Comment (radtriste) [inline Jenkinsfile.release]: don't forget to remove it before merge ;)

Comment (radtriste) [reply Jenkinsfile.release]: same for the test of env

File: Jenkinsfile.release [hunk 2]
```diff
@@ -533,6 +638,10 @@ String getArtifactsRepositoryParam(){
     return params.ARTIFACTS_REPOSITORY
 }
 
+String getOptaPlannerReleaseBranch() {
+    return params.OPTAPLANNER_RELEASE_BRANCH ?: env.OPTAPLANNER_RELEASE_BRANCH
```
Comment (radtriste) [inline Jenkinsfile.release]: this check could be done in initialize part and then only return `env.OPTAPLANNER_RELEASE_BRANCH` here.
wdyt ?

File: Jenkinsfile.release [hunk 3]
```diff
@@ -198,7 +276,8 @@ pipeline {
                     String jobUrl = getJobUrl(EXAMPLES_DEPLOY)
                     def deployProps = jobUrl ? readDeployProperties(jobUrl) : [:]
 
-                    env.EXAMPLES_URI = params.EXAMPLES_URI ?: (deployProps['kogito-examples.pr.source.uri'] ?: "https://github.com/${getGitAuthor()}/kogito-examples")
+                    env.STAGING_REPOSITORY = params.ARTIFACTS_REPOSITORY ?: (deployProps['staging-repo.url'] ?: '')
```
Comment (radtriste) [inline Jenkinsfile.release]: why do you need to retrieve the `STAGING_REPOSITORY` here ?

Comment (rsynek) [reply Jenkinsfile.release]: Good catch, this must have been a slip from conflict resolution.

File: Jenkinsfile.create-release-branches [hunk 4]
```diff
@@ -36,7 +36,8 @@ pipeline {
                         echo "Checkout repo ${repo}"
                         dir(repo){
                             deleteDir()
-                            checkout(githubscm.resolveRepository(repo, getGitAuthor(), 'master', false))
+                            // TODO: return back to 'master' after testing
+                            checkout(githubscm.resolveRepository(repo, getGitAuthor(), 'planner2142', false))
```
Comment (radtriste) [inline Jenkinsfile.create-release-branches]: @rsynek if you are finished with release testing, I guess that one can be reverted :)

File: Jenkinsfile.release [hunk 5]
```diff
@@ -168,6 +208,30 @@ pipeline {
...
+
+                    def buildParams = getDefaultBuildParams()
+                    addSkipTestsParam(buildParams)
+                    addReleaseDeployGitParam(buildParams)
+                    addStringParam(buildParams, 'BUILD_BRANCH_NAME', getOptaPlannerReleaseBranch())
+                    addStringParam(buildParams, 'PROJECT_VERSION', getOptaPlannerVersion())
+                    addStringParam(buildParams, 'KOGITO_VERSION', getProjectVersion())
+                    if (params.ARTIFACTS_REPOSITORY != '') {
+                        addStringParam(buildParams, 'MAVEN_DEPLOY_REPOSITORY', params.ARTIFACTS_REPOSITORY)
+                    }
```
Comment (radtriste) [inline Jenkinsfile.release]:
```suggestion
                    if (getArtifactsRepositoryParam()) {
                        addStringParam(buildParams, 'MAVEN_DEPLOY_REPOSITORY', getArtifactsRepositoryParam())
                    }
```

File: Jenkinsfile.release [hunk 6]
```diff
@@ -198,7 +263,7 @@ pipeline {
                     String jobUrl = getJobUrl(EXAMPLES_DEPLOY)
                     def deployProps = jobUrl ? readDeployProperties(jobUrl) : [:]
 
-                    env.EXAMPLES_URI = params.EXAMPLES_URI ?: (deployProps['kogito-examples.pr.source.uri'] ?: "https://github.com/${getGitAuthor()}/kogito-examples")
+                    env.EXAMPLES_URI = params.EXAMPLES_URI ?: (deployProps['kogito-examples.pr.source.uri'] ?: '')
```
Comment (radtriste) [inline Jenkinsfile.release]: why was this changed ?

**Review by radtriste — CHANGES_REQUESTED** [review]

File: Jenkinsfile.release (hunk 6 above)
Comment (rsynek) [reply Jenkinsfile.release]: Fixed - might be a result of manual conflict resolution.

**Review by radtriste — APPROVED** [review]
